src/Control/Distributed/Process/Internal/CQueue.hs

```diff
@@ -101,7 +109,7 @@ dequeue :: forall m a.
         -> BlockSpec         -- ^ Blocking behaviour
         -> [MatchOn m a]     -- ^ List of matches
         -> IO (Maybe a)      -- ^ 'Nothing' only on timeout
-dequeue (CQueue arrived incoming) blockSpec matchons = mask_ $
+dequeue (CQueue arrived incoming size) blockSpec matchons = mask_ $ decrementJust $
```

Comment:
on a timeout `Nothing` is returned, so we don't decrement then. And because of `mask_` timout may happen only when thread is blocked on `MVar` so we are happy in any case, I suppose.